src/pip/_internal/network/download.py

```diff
@@ -81,7 +81,7 @@ def parse_content_disposition(content_disposition: str, default_filename: str) -
     Parse the "filename" value from a Content-Disposition header, and
     return the default filename if the result is empty.
     """
-    _type, params = cgi.parse_header(content_disposition)
+    _type, params = parse_dict_header(content_disposition)
     filename = params.get("filename")
```

Comment:
```python
    message = Message()
    message['content-type'] = content_disposition
    filename = message.get_param("filename")
 ```

Reply:
Thanks! I finally found the same suggestion buried in a discourse thread from several years ago. Now that I've worked out the nuances of pip's testsuite, I'm a little more comfortable it's a viable solution and will have it up momentarily.

Reply:
For the record, this is also suggested in PEP 594 🙂